freud/box.pyx

```diff
@@ -298,13 +298,14 @@ cdef class Box:
 
         vecs = freud.common.convert_array(
             vecs, vecs.ndim, dtype=np.float32, contiguous=True)
+        images = np.empty_like(vecs, dtype=int)
 
         if vecs.ndim == 1:
-            vecs[:] = self._getImage(vecs)
+            images[:] = self._getImage(vecs)
         elif vecs.ndim == 2:
             for i, vec in enumerate(vecs):
-                vecs[i] = self._getImage(vec)
-        return vecs
+                images[i] = self._getImage(vec)
+        return images
```

Comment:
This function might be a bit more readable if you just made a list `images` and then returned `np.asarray(images)`. Then you could do the 2d case as just a listcomp and the 1d case wouldn't need the colon. But I'll leave it to you whether or not you care enough to make the change.